lib/CArtHandler.cpp

```diff
@@ -871,15 +858,16 @@ void CArtifactInstance::putAt(ArtifactLocation al)
 {
 	assert(canBePutAt(al));
 
-	al.getHolderArtSet()->setNewArtSlot(al.slot, this, false);
+	al.getHolderArtSet()->CArtifactSet::putArtifact(al.slot, this);

```

Comment:
Indended? Why such construct is needed instead of simple `->putArtifact`?

Reply:
Frankly I'd rather have these 3 lines of duplicated code but more clear code than calling specific override of a virtual method.